### DISCUSSION TITLE
Trying to understand correct network configuration for my IIAB environment [DHCP manual override works w/ Ubuntu 22.04+ netplan]

### DISCUSSION THREAD

**Discussion by mbonett** [iiab/iiab#3724] [Q&A] [answered]
Hello,

I have read the IIAB networking materials multiple times. I do not fully understand it for my environment, and it is not  fully working so there is something I am missing. Any information on what I am doing wrong is welcome.

I am trying to set up an IIAB server (Ubuntu Server 22.04) with 2 Ethernet adapters:
1. eth1 -  connected to a switch and with internet access. As I understand it, this is the "WAN" from an IIAB perspective. The network subnet is 10.57.1.x. For this discussion I will call it the WAN network.
2. eth2 - connected to a router. The router broadcasts a wireless network (I call it Box). Nothing else is physically connected to this router other than the IIAB server. 

What I hope to accomplish is:
- Admin tasks and software updates to the IIAB server done through the WAN connection and 192.168.1.x network.
- IIAB users connect to the Box wireless network and access IIAB.

I set the router IP address to 10.10.10.1 and disabled DHCP. I then installed IIAB. The result was the br0 address defined as 10.10.10.10 (as expected), but no address assigned to eth2. The IIAB console can be accessed from the WAN network via eth1 fine. But nothing connected to the Box network can see anything.

I then tried to see if it would make  a difference enabling DHCP on the router (though the does indicate that this is not recommended). This did give a 10.10.10.x address to eth2, but still no access to IIAB via the Box network.

So, what am I missing and doing wrong? A lot of the networking discussion around gateways in the documentations seem to imply the wireless adapter is on the server. Using a connected router, it is not clear to me from the documentation the steps needed to get i working, so that devices on the Box network can access IIAB. For example (tried both ways) should eth2 be connected to the WAN or LAN port of the router? It seems that, with the IIAB server performing DHCP tasks, that eth2 should connect to the router WAN port to assign an address to the router... but I am not sure.  

Any advice or pointers to already existing detailed documentation on this type of setup is appreciated. Thanks in advance.

EDIT: Corrected my WAN network IP information above.

**Comment by holta** [comment]
@mbonett for starters, can you please run `iiab-diagnostics`  and reply with the URL generated?  Thank you!

ASIDE: Did you run `sudo iiab-network` after physically configuring your WiFi router?

9. Optionally, Attach a Wi-Fi Access Point or Ethernet Router (LAN Side)
https://github.com/iiab/iiab/wiki/IIAB-Networking#9-optionally-attach-a-wi-fi-access-point-or-ethernet-router-lan-side

Hopefully this is _not_ needed, but just FYI:

3. Exclude an Interface from br0
https://github.com/iiab/iiab/wiki/IIAB-Networking#3-exclude-an-interface-from-br0

**Reply by mbonett** [reply]
Here is the URL: http://sprunge.us/A1y6Rm

I did run iiab-network while the router was active.

One question about that Wi-Fi access point section - I did read it, but it was not clear which port on the router the IIAB server ethernet port should be connected to (router WAN or LAN).

**Comment by jvonau** [comment]
see https://github.com/iiab/iiab/issues/3385

**Reply by holta** [reply]
@mbonett please read over #3385.

Does that help?

**Comment by jvonau** [comment, marked as answer]
>-IIAB--------------------------------------------------------------------------
-rw-r--r-- 1 root root 120 Mar 27 17:30 /etc/systemd/network/IIAB-Slave-enp3s5.network
                        ...ITS LAST 100 LINES FOLLOW...
>\# /etc/systemd/network/IIAB-Slave-enp3s5.network
[Match]
Name=enp3s5
[Link]
RequiredForOnline=no
[Network]
Bridge=br0

>-rw-r--r-- 1 root root 147 Mar 27 02:52 /etc/netplan/00-installer-config.yaml
                        ...ITS LAST 100 LINES FOLLOW...
\# This is the network config written by 'subiquity'
network:
  ethernets:
    enp2s0:
      dhcp4: true
    enp3s5:
      dhcp4: true
  version: 2




Please # out in 00-installer-config.yaml and reboot.
```
    enp3s5:
      dhcp4: true
```

**Reply by mbonett** [reply]
@jovanu, thank you - commenting out those lines in 00-installer-config.yaml seems to have done the trick!

I have the enp3s5 ethernet port (which I referred to as eth2 in my original post just to simplify things) cabled to the WAN port of the router. After commenting out enp3s5 from 00-installer-config.yaml and rebooting, the router received from the server a 10.10.10.x DHCP address with 10.10.10.10 as the gateway.  Wireless devices that connect to the router wireless network can now access IIAB. Thank you!

Perhaps what also contributed to my confusion is that I am using the router device in router mode. re-reading the documentation and item #3385 makes we wonder if I should have set the router device to AP mode in the first place. I test that - but at least I know I have one mode working. Thank you!

**Reply by mbonett** [reply]
So I tried with the router in AP mode, and it still worked, with the wireless clients now getting the IP subnet addresses supplied by the IIAB server. Now I understand the networking aspect better when one is in gateway mode and an attached router is used on a 2nd ethernet port to provide the wireless network for the IIAB client. Setting the router to AP mode makes things simpler.

For me, a picture is worth a thousand words, so I attached a simple picture as a very summaryfor my situation. 
![ScreenShot071](https://github.com/iiab/iiab/assets/7140289/0632c632-8aa0-47de-a846-d0b08832a4e9)

**Comment by holta** [comment]
@mbonett out of curiosity will you be using Ubuntu 24.04 LTS, e.g. after that's released very shortly in April?

(And if so can we ask you to report back how that works?!)

Ubuntu 24.04 Tips:
https://github.com/iiab/iiab/wiki/IIAB-Platforms#ubuntu-2404

**Reply by mbonett** [reply]
> @mbonett out of curiosity will you be using Ubuntu 24.04 LTS, e.g. after that's released very shortly in April?
> 
> (And if so can we ask you to report back how that works?!)
> 
> Ubuntu 24.04 Tips: https://github.com/iiab/iiab/wiki/IIAB-Platforms#ubuntu-2404

I might... I am busy with some travel the first part of April so the earliest might get to would be around the 15th or 20th. If I am able I will be happy to provide feedback.

**Reply by holta** [reply]
Awesome!

And no rush as Ubuntu 24.04 only becomes "official" on April 25th :smile:
